### DISCUSSION TITLE
How to set Electron file dialog to modal?

### DISCUSSION THREAD

**Discussion by zole40** [eclipse-theia/theia#13037] [Q&A] [answered]
Hello. I want to open the file dialogue on the same screen as the application. Now it always opens on the main screen.
 I found an [issue ](https://github.com/eclipse-theia/theia/issues/10768)and [PR ](https://github.com/eclipse-theia/theia/pull/10769/commits/253fbd45304ac8d43e6f98d85550013a0536e2a4)that made it possible, but the modal property is undefined by default, and I did not find a preference,  a flag or any other way to set it to true. 
Can I change it, or do I have to implement my ElectronFileDialogService?

**Comment by JonasHelming** [comment]
@AlexandraBuzila

**Comment by zole40** [comment]
In the meantime I made a workaround to fix it in our app, but I think there must be a better way.![image](https://github.com/eclipse-theia/theia/assets/10961210/46352182-753a-47a3-86cd-c170e1650096)
![image](https://github.com/eclipse-theia/theia/assets/10961210/16a30d2f-0d8b-4fb8-990b-59d30458e868)

**Comment by AlexandraBuzila** [comment, marked as answer]
Hi @zole40 

There's currently no way to set the modal option via a preference. ~~The expectation is that the callers of the FileDialogService pass the `modal` prop in the arguments, for example:~~

```
this.fileDialogService.showOpenDialog({ title: 'My Open Dialog', canSelectFiles: false, modal: true }, folder)
```

If you want all open and save dialogs created by the service to always be modal, then your workaround is the way to go. 

Hope this helps!

**Reply by AlexandraBuzila** [reply]
I just double checked my own code, it's been a while since I wrote it. 

The initial intention was that if the `modal` prop is undefined, it it still being evaluated as true. So all dialogs should have been modal by default.
This seems to no longer be the case and it's likely a regression. Could you open an issue for this, please?

**Reply by zole40** [reply]
https://github.com/eclipse-theia/theia/issues/13042
